tests/functional/ArchiveTest.php

```diff
@@ -0,0 +1,427 @@
+            ])
+        );
+        $actualArchivePath = $this->getOutput();
+        $this->assertEquals($this->archivePath, $actualArchivePath);
+
+        // Validate database credentials are present in settings.php file.
+        $this->drush(
+            'archive:dump',
+            [],
+            [],
+            null,
+            null,
+            self::EXIT_ERROR
+        );
+        $this->assertStringContainsString(
+            'Found database connection settings',
+            $this->getErrorOutput()
+        );
+    }
+
+    public function testArchiveRestoreCommand(): void
+    {
+        // [info] Copying files from "C:/projects/work/sandbox/archive/code\" to "C:\projects\work\"...
+        // [info] Executing: rsync -akz --stats --progress -v C:/projects/work/sandbox/archive/code\ C:\projects\work\
+        // > The source and destination cannot both be remote.
+        if ($this->isWindows()) {
+            $this->markTestSkipped('The command archive:restore does not work on Windows yet due to an rsync issue.');
+        }
+
+        // Restoring to sqlite fails
+        if ($this->dbDriver() === 'sqlite') {
+            $this->markTestSkipped('The command archive:restore cannot restore to an sqlite database.');
+        }
+
+        // Restore archive from an existing file.
+        $this->assertFalse(is_dir($this->restorePath));
+        $this->drush(
+            'archive:restore',
+            [$this->archivePath],
+            array_diff_key($this->archiveRestoreOptions, ['overwrite' => null])
+        );
+        $this->assertTrue(is_dir($this->restorePath));
+        $this->assertTrue(is_file(Path::join($this->restorePath, 'composer.json')));
+        $this->assertTrue(is_file(Path::join($this->restorePath, 'composer.lock')));
+
+        // Restore archive from an existing file and an existing destination path.
+        $this->drush(
+            'archive:restore',
+            [$this->archivePath],
+            array_diff_key($this->archiveRestoreOptions, ['overwrite' => null]),
+            null,
+            null,
+            self::EXIT_ERROR
+        );
+        $this->assertMatchesRegularExpression(
+            '/Destination path .+ already exists/',
+            str_replace("\n", " ", $this->getErrorOutput())
+        );
+
+        // Restore archive from paths.
+
+        // Restore code.
+        $testFileName = 'test-file-' . mt_rand() . '.txt';
+        file_put_contents(Path::join($this->extractPath, 'code', 'sut', $testFileName), 'foo_bar');
+        $this->drush(
+            'archive:restore',
+            [],
+            array_merge($this->archiveRestoreOptions, [
+                'code' => null,
+                'code-source-path' => Path::join($this->extractPath, 'code'),
+                'destination-path' => $this->restorePath,
+            ])
+        );
+        $this->assertTrue(is_file(Path::join($this->restorePath, 'sut', $testFileName)));
+        $this->installComposerDependencies();
```

Comment:
The tests shouldn't have to call `composer install`; the `archive:restore` command should do it.

Reply:
I asked to remove `composer install` from restore. IMO its out of scope. Either the command should archive the vendor dir, or it shouldnt. Happy to discuss.

Reply:
@weitzman: The comment above is old. @jabxjab's original implementation did not do a `composer install` in `archive:restore`. I believed that `archive:restore` should produce a functional site, so I told him to move the composer install to the command itself, and he did so. After that, you provided your review comment that you preferred it if `archive:restore` did not call `composer install`, so @rwagner00 reverted back to the old tests, which made GitHub display my old comment again.